File: app/src/main/java/hegde/mahesh/avre/database/HistoryDao.java
package hegde.mahesh.avre.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import hegde.mahesh.avre.model.HistoryItem;

import java.util.List;

@Dao
public interface HistoryDao {
    @Query("SELECT * FROM history")
    List<HistoryItem> getAll();

    @Insert
    void insert(HistoryItem item);

    @Query("DELETE FROM history")
    void delete();
}

File: app/src/main/java/hegde/mahesh/avre/database/HistoryDao.kt
package hegde.mahesh.avre.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import hegde.mahesh.avre.model.HistoryItem

@Dao
interface HistoryDao {
    @Query("SELECT * FROM history")
    fun getAll(): List<HistoryItem>

    @Insert
    fun insert(item: HistoryItem?)

    @Query("DELETE FROM history")
    fun deleteAll()
}